src/earn/EarnEnterAmount.tsx

```diff
@@ -110,25 +134,29 @@ function EarnEnterAmount({ route }: Props) {
   const onTokenPickerSelect = () => {
     tokenBottomSheetRef.current?.snapToIndex(0)
     AppAnalytics.track(SendEvents.token_dropdown_opened, {
-      currentTokenId: token.tokenId,
-      currentTokenAddress: token.address,
-      currentNetworkId: token.networkId,
+      currentTokenId: inputToken.tokenId,
+      currentTokenAddress: inputToken.address,
+      currentNetworkId: inputToken.networkId,
     })
   }
 
   const onSelectToken = (token: TokenBalance) => {
-    setToken(token)
+    setInputToken(token)
     tokenBottomSheetRef.current?.close()
     // NOTE: analytics is already fired by the bottom sheet, don't need one here
   }
 
+  // Avoid conditionally calling hooks
```

Comment:
I've talked with @satish-ravi about adjusting these hooks so that we're not calling both on deposit & withdrawals. I'll make a separate PR with these changes to avoid cluttering up this PR.

Reply:
Changed in 5fb98c5cd76d3022bfa8ed993404301465fe64b6!